src/django/api/views/v1/moderation_events.py

```diff
@@ -114,9 +105,85 @@ def patch(self, request, pk=None):
 
         return Response(
             {
-                "message": 'The request body contains '
+                "detail": 'The request body contains '
                 'invalid or missing fields.',
                 "error": [serializer.errors]
             },
             status=status.HTTP_400_BAD_REQUEST
         )
+
+    @handle_errors_decorator
+    def add_production_location(self, request, moderation_id=None):
+        self.__validate_user_permissions(request)
+
+        self.__validate_uuid(moderation_id)
+
+        event = self.__fetch_moderation_event_by_uuid(moderation_id)
+
+        self.__validate_moderation_status(event.status)
+
+        add_production_location = EventApprovalContext(
+            AddProductionLocationStrategy(event)
+        )
+
+        try:
+            item = add_production_location.run_processing()
+        except Exception as error:
+            return self.__handle_processing_error(error)
+
+        return Response(
+            {"os_id": item.facility_id}, status=status.HTTP_201_CREATED
+        )
+
+    @staticmethod
```

Comment:
I'm just not sure if viewSet is a best way to store private or static methods for endpoint.

Reply:
Maybe replace it to utils.py or create base viewset for moderation event that covered it.

Reply:
It will be great, thank you!

Reply:
@roman-stolar I have moved private and static methods to the newly created `ModerationEventsService` class.

Reply:
Thank you!